File: sandbox-runner/patterns/webhooks/java/src/main/java/Activities.java
import io.temporal.activity.ActivityInterface;

@ActivityInterface
public interface Activities {
    String processPayment(Shared.PaymentPayload payload);

    final class Impl implements Activities {
        @Override
        public String processPayment(Shared.PaymentPayload payload) {
            System.out.printf("Processing payment %s for $%.2f%n", payload.paymentId(), payload.amount());
            return String.format("Payment %s processed successfully for $%.2f",
                    payload.paymentId(), payload.amount());
        }
    }
}

File: sandbox-runner/patterns/webhooks/python/activities.py
from temporalio import activity

from shared import PaymentPayload


@activity.defn
async def process_payment(payload: PaymentPayload) -> str:
    activity.logger.info(f"Processing payment {payload.payment_id} for ${payload.amount:.2f}")
    return f"Payment {payload.payment_id} processed successfully for ${payload.amount:.2f}"
